CONTRIBUTING.md

```diff
@@ -1,65 +1,121 @@
+In the past this was organized using Redmine at: [dev.exiv2.org](http://dev.exiv2.org). As of April 2017 the Exiv2 project has moved from svn and Redmine to GitHub and information, such
+as, Wikis etc., will gradually be moved from Redmine to GitHub. User requests on the Redmine forums and the Redmine issue tracking system will be serviced until Exiv2 0.27 is released. 
+After 0.27 is released the Redmine site will be read-only and all feature requests, bug reports etc., must be performed on GitHub. It is, however, preferred to use GitHub for all new 
+feature requests and bug reports to Exiv2.
+
+# 2. General Workflow for Contributing Code #
+
+Code contributions is performed by using *pull requests* (PR) on GitHub. For this to work one first needs to 
+[create a user account on GitHub](https://help.github.com/articles/signing-up-for-a-new-github-account/) if one do not already have one. Once you have a GitHub login:
+
+1. Fork the Exiv2 git repository to your GitHub account by pressing the _Fork_ button at: [https://github.com/Exiv2/exiv2](https://github.com/Exiv2/exiv2) (more details [here](https://guides.github.com/activities/forking/)).
+
+1. Then start a terminal (or use your favorite git GUI app) and clone your copy of the Exiv2 repo:
+
+        $  git clone https://github.com:your-username/exiv2.git
+
+2. Configure the project and build it (CMake is our the preferred configuration system, but not the only one [see [README.md](README.md)]):
+
+        $ cd exiv2
+		$ rm -rf build
+		$ mkdir build
+		$ cd build
+        $ cmake -DCMAKE_CXX_FLAGS="-fsanitize=address -fsanitize=undefined" ..
+		$ cmake --build . --config Release
 
 3. Make sure the tests pass:
-```
-make tests          # Application tests
-./bin/unit_tests    # Unit tests
-```
 
-4. Make your change. Add tests for your change. Make the tests pass.
-5. Create the Pull Request (PR).
-    a. Check if the CI checks pass.
-    b. If your PR lives for a long time, try to avoid to press the button "Update branch" in the Pull Request view. We prefer rebasing in top of master, instead of merging master into branches.
-    c. A project member will review your PR as soon as possible.
+        $ make tests         # Application tests
+        $./bin/unit_tests    # Unit tests
+
+4. Make your change(s), add tests for your change, and make the tests pass.
+5. Create the pull request by pressing the _New pull request_ button on: https://github.com/your-username/exiv2
+6. After the PR has been reviewed by an exiv2 project member you must rebase your repo copy (we use the name exiv2\_upstream as an example below):
+
+        $ git remote add exiv2_upstream https://github.com/Exiv2/exiv2.git
+        $ git pull --rebase exiv2_upstream master
 
-## Pull Request guidelines
+7. and if you are asked to add a patch:
```

Comment:
This happens relatively rarely, so I don't think it is necessary to add it here.

Reply:
I agree with @D4N. And, again, thinking about the scope of the document, I am not sure if it is a good idea to add details about general git usage or code contribution guidelines here.